src/utils/kernel.rs

```diff
@@ -49,6 +49,22 @@ where
         let channel_shape = self.filter.output_shape(channelless_input);
         Tensor::from_shape_vec(channel_shape, output).unwrap()
     }
+
+    pub fn weights(&self) -> Vec<f64> {
+        self.weights.iter().map(|v| v.value()).collect()
+    }
+
+    pub fn bias(&self) -> f64 {
+        self.bias.value()
+    }
+
+    pub fn weights_mut(&mut self) -> &mut Tensor<E> {
+        &mut self.weights
+    }
+
+    pub fn bias_mut(&mut self) -> &mut Value {
+        &mut self.bias
+    }
```

Comment:
Since I suggested using `.value_mut()` in a previous comment can we make do without these methods? Getting just the weights and bias and iterating over each value, although not mutable, should still work since `.value_mut()` does not require `&mut self`. Please lmk if this is possible

Reply:
If I removed these methods, we still need replacements right? Because I can't access a kernel's `weights` or `bias` outside of the struct implementation. Calling `kernel.weights()` and `kernel.biases()` doesn't work because those return `Vec<f64>`.

Reply:
Good insight, might have to make those return tensors then!